#! /usr/bin/env python
# coding=utf-8

import roslib
from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "lane_det"
roslib.load_manifest(PACKAGE)

# 创建一个参数生成器
gen = ParameterGenerator()

# 添加参数说明，便于后续生成界面
# 参数名 类型 等级 参数描述 默认值 最小值 最大值
gen.add("revise_theta",        double_t,        0,        "revise_theta.",            -0.01919,            -0.2,        0.2)
gen.add("revise_y",            double_t,        0,        "revise_y.",                -0.35,            -1.0,        1.0)
gen.add("delta_scale",         double_t,        0,        "delta_scale.",             1.0,            -10.0,        10.0)

# 调用生成器生成config配置文件
# 包名 节点名称 生成文件名
exit(gen.generate(PACKAGE, PACKAGE, "LANE_PANEL"))